### ISSUE TITLE
Yarn workspace issue with node-fetch

### ISSUE THREAD

**Issue by JoviDeCroock** [dealmore/terraform-aws-next-js#93]
Hey all,

Thanks for the great tool, when using this from a `yarn workspace` structured like the following:

```
root
 |  ...otherWorkspaces
 | web --> contains next, ...
```

When I run `cd web && yarn tf-next --skipDownload` and do a `terraform apply` I get the following error in the `PAGE_LAMBDA`:

```
2021-03-30T07:16:17.284Z	7dbd5f95-e964-4bd1-9ea0-ef2ddfd92413	ERROR	Unhandled error during request: Error: Cannot find module 'node-fetch'
Require stack:
- /var/task/node_modules/next/dist/next-server/server/node-polyfill-fetch.js
```

Is there some `noHoist` that I have to do to account for the workspace or me deploying from it?

EDIT: I've tried forcing `node-fetch` as a dep by using `noHoist` on it but now the error has moved on to the next dependency.

**Comment by ofhouse** [comment]
Difficult to say without having a reproduction at hand.
Can you tell where the node-fetch package is installed, after you have run `yarn`?

```
root/
├── node_modules/
│   └── node-fetch/ <-- Is it here
└── web/
    └── node_modules/
        └── node-fetch/ <-- Or is it here
```

Can you also tell me which Next.js version you are using?

Basically dependencies are bundled by a tool called [nft](https://github.com/vercel/nft) which runs a static analysis on the actual code which dependencies are required.
Since it follows the Require.js algorithm it should be able to catch the dependencies in both cases.
You can also check the result by opening the Lambda Zip files in `.next-tf/lambdas/__NEXT_PAGE_LAMBDA_XX.zip` which should contain `node_modules/node-fetch` directory.

**Comment by JoviDeCroock** [comment]
so all the deps are in `root/node_modules` and we're running `yarn tf-next build --skipDownload` in `/web` and we run `terraform apply` in `/web/.aws/deployment`. 

When I remove `web` from the `root pkg.json` and have my `modules` in `web` it works correctly, I can try making a reproduction. Sorry for the vague issue report, will attempt to get it reproduced

**Comment by ofhouse** [comment]
No problem, setup a repro here: https://github.com/ofhouse/repro-terraform-aws-next-js-93
Can confirm that the tracing of dependencies with `--skipDownload` is broken.

Running `yarn` in the workspace root and after that `npx tf-next build` in the `web/` folder.
No `node_modules` are bundled with the Lambda which should not be happening.

**Comment by JoviDeCroock** [comment]
Managed to bypass the issue for now by running `yarn workspace <my-env> build` from the root rather than in the workspace itself. Good workaround for now 😅

**Comment by JoviDeCroock** [comment]
I guess we need to detect whether or not we're in a workspace [here](https://github.com/dealmore/terraform-aws-next-js/blob/main/packages/runtime/src/index.ts#L1109-L1131)

**Comment by ofhouse** [comment]
Yep, worked a bit with [nft](https://github.com/vercel/nft) last week, and `baseDir` must be the root of the workspace: https://github.com/dealmore/terraform-aws-next-js/blob/main/packages/runtime/src/index.ts#L1125

Will try to take a closer look into this today.

**Comment by ofhouse** [comment]
Okay, created a potential fix for this issue in #107.
Need some testing, but I think this should solve the issue.

**Comment by hugheba** [comment]
Was the bug for the missing node_modules folder ever fixed?

I can build the .next-tf folder fine with the command `./node_modules/.bin/tf-next build --skipDownload` but when I open the Lambda zips, they are missing the node_modules folder.
